src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/TokenBucketRateLimiterOptions.cs

```diff
@@ -0,0 +1,93 @@
+        /// Specifies whether token replenishment will be handled by the <see cref="TokenBucketRateLimiter"/> or by another party via <see cref="TokenBucketRateLimiter.TryReplenish"/>.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">When <paramref name="tokenLimit"/>, <paramref name="queueLimit"/>, or <paramref name="tokensPerPeriod"/> are less than 0
+        /// or when <paramref name="replenishmentPeriod"/> is more than 49 days.</exception>
+        public TokenBucketRateLimiterOptions(
+            int tokenLimit,
+            QueueProcessingOrder queueProcessingOrder,
+            int queueLimit,
+            TimeSpan replenishmentPeriod,
+            int tokensPerPeriod,
+            bool autoReplenishment = true)
+        {
+            if (tokenLimit < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(tokenLimit));
+            }
+            if (queueLimit < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(queueLimit));
+            }
+            if (tokensPerPeriod < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(tokensPerPeriod));
+            }
+            if (replenishmentPeriod.TotalDays > 49)
+            {
+                throw new ArgumentOutOfRangeException(nameof(replenishmentPeriod), "Over 49 days is not supported");
+            }
+
+            TokenLimit = tokenLimit;
+            QueueProcessingOrder = queueProcessingOrder;
+            QueueLimit = queueLimit;
+            ReplenishmentPeriod = replenishmentPeriod;
+            TokensPerPeriod = tokensPerPeriod;
+            AutoReplenishment = autoReplenishment;
+        }
+
+        /// <summary>
+        /// Specifies the minimum period between replenishments.
+        /// </summary>
+        public TimeSpan ReplenishmentPeriod { get; }
+
+        /// <summary>
+        /// Specifies the maximum number of tokens to restore each replenishment.
+        /// </summary>
+        public int TokensPerPeriod { get; }
+
+        /// <summary>
+        /// Specified whether the <see cref="TokenBucketRateLimiter"/> is automatically replenishing tokens or if someone else
+        /// will be calling <see cref="TokenBucketRateLimiter.TryReplenish"/> to replenish tokens.
+        /// </summary>
+        public bool AutoReplenishment { get; }
+
+        /// <summary>
+        /// Maximum number of tokens that can be in the bucket at any time.
+        /// </summary>
+        public int TokenLimit { get; }
+
+        /// <summary>
+        /// Determines the behaviour of <see cref="RateLimiter.WaitAsync"/> when not enough resources can be leased.
+        /// </summary>
+        /// <value>
+        /// <see cref="QueueProcessingOrder.OldestFirst"/> by default.
+        /// </value>
+        public QueueProcessingOrder QueueProcessingOrder { get; }
+
+        /// <summary>
+        /// Maximum cumulative token count of queued acquisition requests.
```

Comment:
This description is different than ConcurrencyLimiterOptions:

>        /// Maximum number of permits that can be queued concurrently.

Reply:
This is done on purpose. The API review called out that TokenBucket should refer to permits as tokens where possible.